interview-questions.md

```diff
@@ -7,64 +7,64 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: The console.log() is what lines of code the user wants to appear in the machine or in the console either in the terminal or the console in a browser. It can be useful for finding bugs or to see changes in the code. A return is the end value of some logic like a for loop or an if statement. There is only one return per loop or function.
 
-  Researched answer:
+  Researched answer: First off, console.log() is a function and it is used to print information to the console. The parameter it passes will be shown in the console window or terminal. The return is a call to pass a value back up to whereever the call was made in the code. When a return is used, the function will stop executing and return the specified value.
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: An argument is what the function accepts and will be used to replace some variable inside the function's block of code. A function will have parameters that act as place holders. These parameters have no actual value and can represent any data type the user inputs as arguments. When the function is invoked, the user's input(s) is the argument which can be any data type.
 
-  Researched answer:
+  Researched answer: Arguments are parameter values in a function. They are the real values passed and recieved by the function when they invoke. Not all functions require arguments. However, if a function requires a parameter, missing arguments are set to undefined.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: There are two roles in pair programming: driver and navigator. The driver is the one who uses the mouse and keyboard. They are in charge of writing the code in the text editor. They listen to their partner to guide them through the problem. They may also provide suggestions to help their partner through the problem. The navigator is the one explaining how to solve the problem. It is their job to effectively communicate how to write the code or explain their logic. They can also help by looking up additional information on the internet to assist in finding a solution to their code. Both partners need to communicate with each other, especially when there is a misunderstanding or if either persons feel stuck.
```

Comment:
Totally! I'd maybe just mention how the roles can be swapped out over given time periods as well, in common practice.